utils/setStatus.js

```diff
@@ -1,26 +1,27 @@
 'use strict';
 
 const statusCodes = require('http').STATUS_CODES
+    , isUndefined = require('./tools').isUndefined
     , setStatus = () => {
         return function (status) {
             /* eslint-disable no-invalid-this */
             const that = this;
             /* eslint-enable no-invalid-this */
-            let k;
 
-            if (status in statusCodes) {
+            if (!isUndefined(statusCodes[status])) {
                 that.statusCode = parseInt(status, 10);
                 that.statusMessage = statusCodes[status];
-                return;
+                return true;
             }
 
-            for (k in statusCodes) {
+            return Object.keys(statusCodes).some((k) => {
                 if (statusCodes[k].toLowerCase() === status.toLowerCase()) {
                     that.statusCode = parseInt(k, 10);
```

Comment:
Actually, I need the `k` from the function in `some()` to update those values. I can create a new variable, store k in that and then use the if condition, but it seems like going passive voice when you can do it directly. 

Can you explain what side-effects could occur in it?